File: 2020/1.1.rb
require_relative 'helpers.rb'

module DayOnePartOne
  module_function

  def solve(file)
    inputs = prepare_inputs(file)
    find_2020_sum(inputs)
  end

  def find_2020_sum(inputs)
    inputs.each do |a|
      inputs.each do |b|
        return a * b if a + b == 2020
      end
    end
  end

  def prepare_inputs(file)
    Helpers
      .split_inputs_by_line(Helpers.read_file(file))
      .map(&:to_f)
  end
end

# puts DayOnePartOne.solve(__dir__ + '/1test.txt')
puts DayOnePartOne.solve(__dir__ + '/1.txt')

File: 2021/1.1.php
<?php


require "Helpers.php";

class DayOnePartOne
{
    function solve($filename)
    {
        $inputs = $this::prepareInputs($filename);
        echo $this->countIncreasing($inputs) . "\n";
    }

    static function prepareInputs($filename)
    {
        return array_map("intval", Helpers::splitInputs($filename));
    }

    function countIncreasing($inputs)
    {
        $previous = $inputs[0];
        $total = 0;
        foreach ($inputs as $input) {
            $total += $input > $previous ? 1 : 0;
            $previous = $input;
        }
        return $total;
    }
}

$class = new DayOnePartOne();
// $class-->solve("/2021/1test.txt");
$class->solve("/2021/1.txt");
